cmsl1t/config.py

```diff
@@ -9,19 +9,35 @@
 import logging
 from cmsl1t.utils import module
 from copy import deepcopy
+import re
 
 
 logger = logging.getLogger(__name__)
 TODAY = datetime.now().timetuple()
 
 
 def get_unique_out_dir(outdir=None, revision=1):
-    full_outdir = outdir + "-rev_{rev}".format(rev=revision)
+    full_outdir = outdir + "-v{rev}".format(rev=revision)
     if os.path.isdir(full_outdir):
         return get_unique_out_dir(outdir, revision + 1)
     return full_outdir
 
 
+def get_last_version_of(outdir):
```

Comment:
`get_unique_out_dir` is a recursive function, which is called without the revision number being set.  And so similarly the caller of this method doesn't know what the value of `revision` is (in your example code, you don't actually pass in this value).  Thus the `get_last_version_of` function is also unaware of the current version and just searches for the last version.  If you specify a hist file(s) this behaviour is essentially disabled since this indicates implicitly the output histogram directory.